src/EditorFeatures/Core.Wpf/InlineRename/UI/Dashboard/RenameDashboard.xaml.cs

```diff
@@ -318,13 +325,21 @@ private void CloseButton_Click(object sender, RoutedEventArgs e)
         }
 
         private void Apply_Click(object sender, RoutedEventArgs e)
-            => Commit();
+        {
+            var token = _listener.BeginAsyncOperation(nameof(Apply_Click));
+            // CommitAsync catch & report all the exceptions.
+            _ = CommitAsync().CompletesAsyncOperation(token);
+        }
 
-        private void Commit()
+        /// <remarks>
+        /// CommitAsync is non-throwing.
+        /// </remarks>
+        private async Task CommitAsync()
         {
             try
             {
-                _model.Session.Commit();
+                //.ConfigureAwait(true) because UI thread is need to change focus
+                await _model.Session.CommitAsync(previewChanges: false).ConfigureAwait(true);
                 _textView.VisualElement.Focus();
             }
             catch (NotSupportedException ex)
```

Comment:
this logic looks dplicated with the RenameCommandHandler.  can it move to a central location?  keep tracking of how different everything does things is hard :)